cli.py

```diff
@@ -1,5 +1,31 @@
 import click
+from ml.base import MLConfig
+from ml.base.estimator import MLEstimator
 
-
+@click.group()
 def main():
     pass
+
+@click.command()
+@click.option('--config', help='The config file')
+def train(config):
+    settings = MLConfig.from_file(config)
+    # The command line action should override the one in config
+    settings.global_config.mode = 'train'
+    model = MLEstimator.from_config(settings)
```

Comment:
don't use the name model, we try to use `estimator` and `graph` to separate the two kinds of notions of `model`. Here it should be `model`